[libtorch] Build failue

Package: libtorch[core,xnnpack]:x64-linux -> 1.12.1#4

**Host Environment**

- Host: x64-linux
- Compiler: GNU 13.2.1
-    vcpkg-tool version: 2023-03-29-664f8bb619b752430368d0f30a8289b761f5caba
    vcpkg-scripts version: 50bffcc62 2023-12-14 (5 hours ago)

**To Reproduce**

`vcpkg install libtorch`

**Failure logs**

```
-- Note: libtorch only supports dynamic library linkage. Building dynamic library.
-- Using cached pytorch-pytorch-v1.12.1.tar.gz.
-- Cleaning sources at /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /opt/vcpkg/downloads/pytorch-pytorch-v1.12.1.tar.gz
-- Applying patch pytorch-pr-85958.patch
-- Applying patch fix-cmake.patch
-- Applying patch fix-fbgemm-include.patch
-- Applying patch fix-c10-glog.patch
-- Applying patch use-flatbuffers2.patch
-- Applying patch fix-windows.patch
-- Applying patch fix_werror.patch
-- Using source at /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean
-- Using flatc: /opt/vcpkg/installed/x64-linux/tools/flatbuffers/flatc
-- Using protoc: /opt/vcpkg/installed/x64-linux/tools/protobuf/protoc
-- Setting up python virtual environmnent...
-- Installing python packages: typing-extensions;pyyaml;numpy
-- Setting up python virtual environmnent...finished.
-- Using Python3: /opt/vcpkg/buildtrees/libtorch/x64-linux-venv/bin/python
-- Found external ninja('1.11.1').
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /opt/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake --build . --config Debug --target torch_cpu -- -v -j33
    Working Directory: /opt/vcpkg/buildtrees/libtorch/x64-linux-dbg
    See logs for more information:
      /opt/vcpkg/buildtrees/libtorch/build-torch_cpu-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  ports/libtorch/portfile.cmake:158 (vcpkg_cmake_build)
  scripts/ports.cmake:170 (include)



```
<details><summary>/opt/vcpkg/buildtrees/libtorch/build-torch_cpu-x64-linux-dbg-out.log</summary>

```
Change Dir: '/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg'

Run Build Command(s): /usr/bin/ninja -v -v -j33 torch_cpu
[1/1053] /usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/core/AutogradState.cpp.o -MF c10/CMakeFiles/c10.dir/core/AutogradState.cpp.o.d -o c10/CMakeFiles/c10.dir/core/AutogradState.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/AutogradState.cpp
[2/1053] /usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/core/Allocator.cpp.o -MF c10/CMakeFiles/c10.dir/core/Allocator.cpp.o.d -o c10/CMakeFiles/c10.dir/core/Allocator.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/Allocator.cpp
...
Skipped 64 lines
...

[67/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/embedding_lookup_avx2.cc
[68/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_fused_8bit_rowwise_idx_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_fused_8bit_rowwise_idx_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_fused_8bit_rowwise_idx_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/embedding_lookup_fused_8bit_rowwise_idx_avx2.cc
[69/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/typed_axpy_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/typed_axpy_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/typed_axpy_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/typed_axpy_avx2.cc
[70/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_idx_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_idx_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/embedding_lookup_idx_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/embedding_lookup_idx_avx2.cc
[71/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/lstm_unit_cpu_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/lstm_unit_cpu_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/lstm_unit_cpu_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/lstm_unit_cpu_avx2.cc
[72/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/math_cpu_avx2.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/math_cpu_avx2.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx2.dir/math_cpu_avx2.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/math_cpu_avx2.cc
[73/1053] /usr/bin/ccache /usr/bin/c++ -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -mavx512f -mavx512dq -mavx512vl -mavx2 -mfma -mavx -mf16c -MD -MT caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx512.dir/common_avx512.cc.o -MF caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx512.dir/common_avx512.cc.o.d -o caffe2/perfkernels/CMakeFiles/Caffe2_perfkernels_avx512.dir/common_avx512.cc.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/caffe2/perfkernels/common_avx512.cc
[74/1053] /usr/bin/ccache /usr/bin/c++ -DONNX_ML=1 -DTORCH_API="" -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -MD -MT caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/torch.pb.cc.o -MF caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/torch.pb.cc.o.d -o caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/torch.pb.cc.o -c /opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/caffe2/proto/torch.pb.cc
[75/1053] /usr/bin/ccache /usr/bin/c++ -DONNX_ML=1 -DTORCH_API="" -DUSE_PTHREADPOOL -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -std=gnu++14 -MD -MT caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/caffe2.pb.cc.o -MF caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/caffe2.pb.cc.o.d -o caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/caffe2.pb.cc.o -c /opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/caffe2/proto/caffe2.pb.cc
[76/1053] : && /opt/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E rm -f lib/libcaffe2_protos.a && /usr/bin/ar qc lib/libcaffe2_protos.a  caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/torch.pb.cc.o caffe2/proto/CMakeFiles/Caffe2_PROTO.dir/caffe2.pb.cc.o && /usr/bin/ranlib lib/libcaffe2_protos.a && :
[77/1053] /usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o -MF c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o.d -o c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/MathConstants.cpp
FAILED: c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o -MF c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o.d -o c10/CMakeFiles/c10.dir/util/MathConstants.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/MathConstants.cpp
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/MathConstants.h:4,
                 from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/MathConstants.cpp:8:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:1: warning: inline variables are only available with ‘-std=c++17’ or ‘-std=gnu++17’ [-Wc++17-extensions]
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      | ^~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:44: error: ‘uint16_t’ was not declared in this scope
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      |                                            ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:9:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
    8 | #include <cstring>
  +++ |+#include <cstdint>
    9 | 
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: error: ‘uint16_t’ does not name a type
   36 | inline C10_HOST_DEVICE uint16_t bits_from_f32(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: error: ‘uint16_t’ does not name a type
   51 | inline C10_HOST_DEVICE uint16_t round_to_nearest_even(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: error: ‘uint16_t’ does not name a type
   74 |   uint16_t x;
      |   ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h: In constructor ‘constexpr c10::BFloat16::BFloat16(short unsigned int, from_bits_t)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:89:9: error: class ‘c10::BFloat16’ does not have any field named ‘x’
   89 |       : x(bits){};
      |         ^
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:101:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In constructor ‘c10::BFloat16::BFloat16(float)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:3: error: ‘x’ was not declared in this scope
   20 |   x = detail::round_to_nearest_even(value);
      |   ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:15: error: ‘round_to_nearest_even’ is not a member of ‘c10::detail’
   20 |   x = detail::round_to_nearest_even(value);
      |               ^~~~~~~~~~~~~~~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In member function ‘c10::BFloat16::operator float() const’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:32: error: ‘x’ was not declared in this scope
   29 |   return detail::f32_from_bits(x);
      |                                ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:33: error: ‘c10::detail::f32_from_bits’ cannot be used as a function
   29 |   return detail::f32_from_bits(x);
      |                                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator|(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator^(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator&(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |                 ^
[78/1053] /usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o -MF c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o.d -o c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/TypeCast.cpp
FAILED: c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o -MF c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o.d -o c10/CMakeFiles/c10.dir/util/TypeCast.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/TypeCast.cpp
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/ScalarType.h:3,
                 from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/TypeCast.h:2,
                 from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/TypeCast.cpp:1:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:1: warning: inline variables are only available with ‘-std=c++17’ or ‘-std=gnu++17’ [-Wc++17-extensions]
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      | ^~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:44: error: ‘uint16_t’ was not declared in this scope
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      |                                            ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:9:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
    8 | #include <cstring>
  +++ |+#include <cstdint>
    9 | 
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: error: ‘uint16_t’ does not name a type
   36 | inline C10_HOST_DEVICE uint16_t bits_from_f32(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: error: ‘uint16_t’ does not name a type
   51 | inline C10_HOST_DEVICE uint16_t round_to_nearest_even(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: error: ‘uint16_t’ does not name a type
   74 |   uint16_t x;
      |   ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h: In constructor ‘constexpr c10::BFloat16::BFloat16(short unsigned int, from_bits_t)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:89:9: error: class ‘c10::BFloat16’ does not have any field named ‘x’
   89 |       : x(bits){};
      |         ^
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:101:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In constructor ‘c10::BFloat16::BFloat16(float)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:3: error: ‘x’ was not declared in this scope
   20 |   x = detail::round_to_nearest_even(value);
      |   ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:15: error: ‘round_to_nearest_even’ is not a member of ‘c10::detail’
   20 |   x = detail::round_to_nearest_even(value);
      |               ^~~~~~~~~~~~~~~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In member function ‘c10::BFloat16::operator float() const’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:32: error: ‘x’ was not declared in this scope
   29 |   return detail::f32_from_bits(x);
      |                                ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:33: error: ‘c10::detail::f32_from_bits’ cannot be used as a function
   29 |   return detail::f32_from_bits(x);
      |                                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator|(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator^(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator&(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |                 ^
[79/1053] /usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -MF c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o.d -o c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/DefaultDtype.cpp
FAILED: c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DC10_BUILD_MAIN_LIB -DFMT_HEADER_ONLY=1 -DGFLAGS_IS_A_DLL=0 -DONNX_ML=1 -DSUPPORTS_BACKTRACE=1 -DUSE_PTHREADPOOL -Dc10_EXPORTS -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg/aten/src -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/aten/src -I/opt/vcpkg/buildtrees/libtorch/x64-linux-dbg -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean -I/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/.. -isystem /opt/vcpkg/installed/x64-linux/include -isystem /opt/vcpkg/installed/x64-linux/include/eigen3 -fPIC -Wno-deprecated -fvisibility-inlines-hidden -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -g -fno-omit-frame-pointer -O0 -std=gnu++14 -fPIC -fvisibility=hidden -MD -MT c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -MF c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o.d -o c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -c /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/DefaultDtype.cpp
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/ScalarType.h:3,
                 from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/DefaultDtype.h:3,
                 from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/core/DefaultDtype.cpp:1:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:1: warning: inline variables are only available with ‘-std=c++17’ or ‘-std=gnu++17’ [-Wc++17-extensions]
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      | ^~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:17:44: error: ‘uint16_t’ was not declared in this scope
   17 | inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
      |                                            ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:9:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
    8 | #include <cstring>
  +++ |+#include <cstdint>
    9 | 
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: error: ‘uint16_t’ does not name a type
   36 | inline C10_HOST_DEVICE uint16_t bits_from_f32(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:36:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: error: ‘uint16_t’ does not name a type
   51 | inline C10_HOST_DEVICE uint16_t round_to_nearest_even(float src) {
      |                        ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:51:24: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: error: ‘uint16_t’ does not name a type
   74 |   uint16_t x;
      |   ^~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:74:3: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h: In constructor ‘constexpr c10::BFloat16::BFloat16(short unsigned int, from_bits_t)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:89:9: error: class ‘c10::BFloat16’ does not have any field named ‘x’
   89 |       : x(bits){};
      |         ^
In file included from /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16.h:101:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In constructor ‘c10::BFloat16::BFloat16(float)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:3: error: ‘x’ was not declared in this scope
   20 |   x = detail::round_to_nearest_even(value);
      |   ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:20:15: error: ‘round_to_nearest_even’ is not a member of ‘c10::detail’
   20 |   x = detail::round_to_nearest_even(value);
      |               ^~~~~~~~~~~~~~~~~~~~~
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In member function ‘c10::BFloat16::operator float() const’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:32: error: ‘x’ was not declared in this scope
   29 |   return detail::f32_from_bits(x);
      |                                ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:29:33: error: ‘c10::detail::f32_from_bits’ cannot be used as a function
   29 |   return detail::f32_from_bits(x);
      |                                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator|(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:102:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  102 |   a.x = a.x | b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator^(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:107:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  107 |   a.x = a.x ^ b.x;
      |                 ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h: In function ‘c10::BFloat16& c10::operator&(BFloat16&, const BFloat16&)’:
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:5: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |     ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:11: error: ‘struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |           ^
/opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean/c10/util/BFloat16-inl.h:112:17: error: ‘const struct c10::BFloat16’ has no member named ‘x’
  112 |   a.x = a.x & b.x;
      |                 ^
[80/1053] cd /opt/vcpkg/buildtrees/libtorch/src/v1.12.1-e1cb316ec6.clean && /opt/vcpkg/buildtrees/libtorch/x64-linux-venv/bin/python tools/setup_helpers/generate_code.py --native-functions-path aten/src/ATen/native/native_functions.yaml --tags-path aten/src/ATen/native/tags.yaml   --force_schema_registration --gen_lazy_ts_backend --per_operator_headers
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

--- a/ports/libtorch/missing_include.patch
+++ b/ports/libtorch/missing_include.patch
@@ -1,0 +1,57 @@
+diff --git a/aten/src/ATen/core/dynamic_type.h b/aten/src/ATen/core/dynamic_type.h
+index a84644ddde0..b1b6b2c9ecc 100644
+--- a/aten/src/ATen/core/dynamic_type.h
++++ b/aten/src/ATen/core/dynamic_type.h
+@@ -1,5 +1,6 @@
+ #pragma once
+ 
++#include <cstdint>
+ #include <memory>
+ #include <type_traits>
+ 
+diff --git a/c10/util/BFloat16.h b/c10/util/BFloat16.h
+index 1ada02bba1c..f6f9fc65e0e 100644
+--- a/c10/util/BFloat16.h
++++ b/c10/util/BFloat16.h
+@@ -3,6 +3,7 @@
+ // Defines the bloat16 type (brain floating-point). This representation uses
+ // 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+ 
++#include <cstdint>
+ #include <c10/macros/Macros.h>
+ #include <cmath>
+ #include <cstring>
+diff --git a/torch/csrc/jit/passes/quantization/quantization_type.h b/torch/csrc/jit/passes/quantization/quantization_type.h
+index ea5ca10b15a..c704f9c3037 100644
+--- a/torch/csrc/jit/passes/quantization/quantization_type.h
++++ b/torch/csrc/jit/passes/quantization/quantization_type.h
+@@ -1,5 +1,6 @@
+ #pragma once
+ #include <ostream>
++#include <cstdint>
+ 
+ namespace torch {
+ namespace jit {
+diff --git a/torch/csrc/jit/runtime/logging.cpp b/torch/csrc/jit/runtime/logging.cpp
+index 66712990368..b326f587bc8 100644
+--- a/torch/csrc/jit/runtime/logging.cpp
++++ b/torch/csrc/jit/runtime/logging.cpp
+@@ -1,6 +1,7 @@
+ #include <torch/csrc/jit/runtime/logging.h>
+ 
+ #include <atomic>
++#include <stdexcept>
+ #include <mutex>
+ #include <unordered_map>
+ 
+diff --git a/torch/csrc/lazy/core/multi_wait.cpp b/torch/csrc/lazy/core/multi_wait.cpp
+index 6b9933518e0..db0742b32a3 100644
+--- a/torch/csrc/lazy/core/multi_wait.cpp
++++ b/torch/csrc/lazy/core/multi_wait.cpp
+@@ -1,5 +1,6 @@
+ #include <torch/csrc/lazy/core/multi_wait.h>
+ 
++#include <stdexcept>
+ #include <chrono>
+ #include <exception>
+ 

--- a/ports/libtorch/portfile.cmake
+++ b/ports/libtorch/portfile.cmake
@@ -7,6 +7,7 @@ vcpkg_from_github(
     SHA512 afeb551904ebd9b5901ae623a98eadbb3045115247cedf8006a940742cfad04e5ce24cfaf363336a9ed88d7ce6a4ac53dbb6a5c690aef6efdf20477c3a22c7ca
     HEAD_REF master
     PATCHES
+        missing_include.patch
         pytorch-pr-85958.patch # https://github.com/pytorch/pytorch/pull/85958
         fix-cmake.patch
         fix-fbgemm-include.patch

--- a/ports/libtorch/vcpkg.json
+++ b/ports/libtorch/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libtorch",
   "version": "1.12.1",
-  "port-version": 4,
+  "port-version": 5,
   "description": "Tensors and Dynamic neural networks in Python with strong GPU acceleration",
   "homepage": "https://pytorch.org/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4918,7 +4918,7 @@
     },
     "libtorch": {
       "baseline": "1.12.1",
-      "port-version": 4
+      "port-version": 5
     },
     "libtorrent": {
       "baseline": "2.0.9",

--- a/versions/l-/libtorch.json
+++ b/versions/l-/libtorch.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "20a2f323fb33dab29315fd18217e263ade2a9bd6",
+      "version": "1.12.1",
+      "port-version": 5
+    },
+    {
       "git-tree": "de2d4bfd4a26268b29c3be02790c16d23ecd3d5e",
       "version": "1.12.1",
       "port-version": 4

